developer-extension/src/background/domain/logEventsFromRequests.ts

```diff
@@ -21,25 +25,38 @@ chrome.webRequest.onBeforeRequest.addListener(
     if (!info.requestBody!.raw) {
       return
     }
-
+    const newEvents: StoredEvent[] = []
     for (const rawBody of info.requestBody!.raw) {
       if (rawBody.bytes) {
         const decodedBody = decoder.decode(rawBody.bytes)
         for (const rawEvent of decodedBody.split('\n')) {
           const event = sortProperties(JSON.parse(rawEvent))
+          newEvents.push(event as StoredEvent)
           void chrome.tabs.executeScript(info.tabId, {
             code: `console.info("Browser-SDK:", ${JSON.stringify(intake)}, ${JSON.stringify(event)});`,
           })
         }
       }
     }
+    storeEvents(newEvents, info.tabId)
   },
   {
     urls: intakeUrlPatterns,
   },
   ['requestBody']
 )
 
+function storeEvents(newEvents: StoredEvent[], tabId: number) {
+  const previousEvents = Object.prototype.hasOwnProperty.call(store.local, tabId) ? store.local[tabId].events : []
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Using the store for storing events isn't great. The whole `store.local` concept is especially not great, because everything will persist forever. We should think of a better design in the future.

Reply:
Yes clearly